File: metronic_spring_v8.2.3/spring/metronic_spring_v8.2.3_demo1/spring_demo1/starterkit/src/main/java/com/theme/starterkit/controllers/AuthController.java
package com.theme.starterkit.controllers;

import com.theme.starterkit.libs.KTTheme;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;

@Controller
public class AuthController {
    @Autowired
    private KTTheme theme;

    @ModelAttribute
    public void init() {
        theme.setLayout("auth");
        theme.initLayout();
    }

    @GetMapping("/new-password")
    public String newPassword(){
        theme.addJavascriptFile("js/custom/authentication/reset-password/new-password.js");
        return theme.getPageView("auth", "new-password");
    }

    @GetMapping("/reset-password")
    public String resetPassword(){
        theme.addJavascriptFile("js/custom/authentication/reset-password/reset-password.js");
        return theme.getPageView("auth", "reset-password");
    }

    @GetMapping("/sign-in")
    public String signIn(){
        theme.addJavascriptFile("js/custom/authentication/sign-in/general.js");
        return theme.getPageView("auth", "sign-in");
    }

    @GetMapping("/sign-up")
    public String signUp(){
        theme.addJavascriptFile("js/custom/authentication/sign-up/general.js");
        return theme.getPageView("auth", "sign-up");
    }
}

File: metronic_asp.net-core_v8.2.3/asp.net-core/metronic_asp.net-core_v8.2.3/metronic_asp.net-core_v8.2.3/Starterkit/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using Starterkit._keenthemes.libs;

namespace Starterkit.Controllers;

public class AuthController : Controller
{
    private readonly ILogger<DashboardsController> _logger;
    private readonly IKTTheme _theme;

    public AuthController(ILogger<DashboardsController> logger, IKTTheme theme)
    {
        _logger = logger;
        _theme = theme;
    }

    [HttpGet("/signin")]
    public IActionResult SignIn()
    {
        return View(_theme.GetPageView("Auth", "SignIn.cshtml"));
    }

    [HttpGet("/signup")]
    public IActionResult SignUp()
    {
        return View(_theme.GetPageView("Auth", "SignUp.cshtml"));
    }

    [HttpGet("/reset-password")]
    public IActionResult ResetPassword()
    {
        return View(_theme.GetPageView("Auth", "ResetPassword.cshtml"));
    }

    [HttpGet("/new-password")]
    public IActionResult NewPassword()
    {
        return View(_theme.GetPageView("Auth", "NewPassword.cshtml"));
    }    
}
